Votiro sentinel solutions

sambhrant-metron opened the pull request:
Change(s):
   - Votiro Solution.

   Reason for Change(s):
   - Integration of Votiro v1.0.0

   Version Updated:
   - Yes

   Testing Completed:
   - Yes
   Checked that the validations are passing and have addressed any issues that are present:
   - Yes

v-mchatla commented:
Hi @sambhrant-metron,
Can you please create a package with 2.0.0 version using the V2 version of the [tool](https://github.com/Azure/Azure-Sentinel/tree/master/Tools/Create-Azure-Sentinel-Solution/V2).
Thanks

sambhrant-metron commented:
hi @v-mchatla,
This package is already created using V2 version tool.
![Screenshot from 2022-12-16 10-17-04](https://user-images.githubusercontent.com/111038486/208024672-5baec154-a91f-4c05-8075-f73711a5a8cd.png)

Please let me know if any changes are needed.

v-mchatla commented:
Hi @sambhrant-metron,
Thanks for the confirmation. Please update the version in input file to 2.0.0 and recreate the package.
Please create Data folder and place your input file in that. Please refer this [folder](https://github.com/Azure/Azure-Sentinel/tree/master/Solutions/Bitglass) structure we follow.
Please revert the changes from this [path](https://github.com/Azure/Azure-Sentinel/tree/master/Tools/Create-Azure-Sentinel-Solution/V2/input)
Thanks

sambhrant-metron commented:
hi @v-mchatla,
I will make the necessary changes.
May close the Pull Request and  create a new PR after making.
Thanks

v-mchatla commented:
> hi @v-mchatla, I will make the necessary changes. May close the Pull Request and create a new PR after making. Thanks

Hi @sambhrant-metron,
Thanks for the update. 
Thanks

sambhrant-metron commented:
Hi @v-mchatla,
I have made the changes.
Please let me know in-case of any issue.
Thanks

sambhrant-metron commented:
@microsoft-github-policy-service agree [company="Votiro"]

sambhrant-metron commented:
@microsoft-github-policy-service agree company="Votiro"

v-mchatla commented:
Hi @sambhrant-metron,
Thanks for the updated package. I will review and share the feedback.
Thanks

v-mchatla commented:
Hi @aprakash13,
Can you please review the analytic rules and share your feedback.
Thanks

v-mchatla commented:
Hi @sambhrant-metron,
I have reviewed the changes, requesting you to fix the following things.
1. Data File: You can refer this sample [document](https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/SlashNext/Data/Solution_SlashNext.json)
         - Can you please modify the logo path it should be Azure Sentinel master log folder path. It will get displayed once the PR gets merged to master.
         - Author details 'Partner' is generic, can you add your organization or team name so that it will be more specific.
         - Description should give a brief idea of what the solution is and how it helps the users. Can you please add clear description.
 2. Can you please share the sample data in .csv file so that I can dump the data to my workspace and test the parser, analytic rules.
 3. Azure Sentinel instances can be replaced by Microsoft Sentinel.
 4. Please add "dataTypesDependencies": [], in workbook metadata.json file
Thanks

sambhrant-metron commented:
hi @v-mchatla,
I have made all the changes except for the "sample data" file, because the logs for which the parser is built is not in CSV format, rather its in CEF syslog format. The text formatted file is already part of the Solution.
Please let me know if that works.
Thanks.

v-mchatla commented:
Hi @sambhrant-metron,
Thanks for addressing the comments. Is it possible for you to share the sample data in json format?
Thanks

sambhrant-metron commented:
hi @v-mchatla,
Thank you for reviewing the files, but the sample data can not be provided in json format.
Please let me know in case of any other changes,
Thanks.

v-mchatla commented:
Hi @sambhrant-metron,
Let me check with team if there is any other way that we can test this and get back to you.
Thanks

v-mchatla commented:
Hi @sambhrant-metron,
Please expect some delay in the process as we are occupied with priority work.
Thanks

v-mchatla commented:
Hi @sambhrant-metron,
Please expect some delay in the process as we are occupied with priority work.
Thanks

v-atulyadav commented:
Hi @sambhrant-metron,
Please expect some delay in the process, we will update you ASAP. Thanks.

v-mchatla commented:
Hi @suhelajjaman,
I'm working with the team on ingesting .txt formatted sample data to Log Analytics for testing. I will keep you posted.
Thanks

v-atulyadav commented:
Looking into this

v-mchatla commented:
Hi @sambhrant-metron,
Sorry for the delay in response. I have dumped the data into logAnalytics and verified the parser. I will verify the other content and provide you the update by today EOD.
Thanks

v-mchatla commented:
Hi @sambhrant-metron,
Sorry for the delay in response. Only Anlytic Rules review is pending as those are pointing to custom tables, I will get back to you once done with the review.
Thanks

v-mchatla commented:
Hi @sambhrant-metron,
Can you please help me in understanding the below queries.
1. Why are you running the analytic rules on top of Votiro_CL
2. How you are populating the data into the table Votiro_CL.
Thanks

v-mchatla commented:
> Hi @sambhrant-metron, Can you please help me in understanding the below queries.
> 
> 1. Why are you running the analytic rules on top of Votiro_CL
> 2. How you are populating the data into the table Votiro_CL.
>    Thanks

Hi @sambhrant-metron,
It would be great if you could help me with the above queries.
Thanks

sambhrant-metron commented:
hi @v-mchatla,
Let me check the DetectionTemplateSchemaValidation issue and fix that.
Then I can help can you with the quries.
Thanks.

sambhrant-metron commented:
hi @v-mchatla ,
Here are the changes for your queries.
1. Why are you running the analytic rules on top of Votiro_CL
    - It should run on VotiroEvents
    - Added this additional table for validation - but this should be named VotiroEvents table
2. How you are populating the data into the table Votiro_CL
    - Data coming from the Data Connector should populate this table, I have renamed it to VotiroEvents
Hope this helps, and please let me know incase of any query.
Thanks.

v-mchatla commented:
Hi @sambhrant-metron,
Thanks for the details. I have verified the analytic rules and those are getting created successfully. 
Thanks

v-mchatla commented:
Hi @devikamehra,
Can you please review the connector and analytic Rules and share your feedback.
Thanks

v-mchatla commented:
Hi @devikamehra,
Can you please review the connector and analytic Rules and share your feedback.
Thanks

v-mchatla commented:
Hi @devikamehra,
Can you please review the connector and analytic Rules and share your feedback.
Thanks

devikamehra commented on Solutions/Votiro/Data Connectors/VotiroEvents.json:
```diff
@@ -0,0 +1,132 @@
+            "name": "Votiro"
+        },
+        "support": {
+            "tier": "Type of support for content item: microsoft | developer | community",
```
Update the value of all these fields.

sambhrant-metron replied:
hi @devikamehra,
We have changed as per requested.

devikamehra commented on Solutions/Votiro/Parser/VotiroEvents.txt:
```diff
@@ -0,0 +1,27 @@
+﻿CommonSecurityLog
+| parse AdditionalExtensions with 
+    "companyName=" CompanyName:string 
```
Use parse-kv function to parse all the fields in one go. Fields can be renamed later.
Current implementation would fail if even one attribute is missing in between.

sambhrant-metron replied:
hi @devikamehra,
We have changed as per requested.

devikamehra commented on Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml:
```diff
@@ -0,0 +1,28 @@
+id: 17bf3780-ae0d-4cd9-a884-5df8b687f3f5
+name: Votiro - File Blocked from Connector
+description: |
+ 'If the syslog message includes-Blocked under  -Sanitization result-field  and false under -password protected- field and null under -from- field  create an alert with the following message:[file name] with hash [file hash] that was sent from connector [connector name] was blocked by Votiro due to Policy [policy name], see more detail in the following link [incident url]'
```
Description should explain the intend of the rule in plain English instead of explaining each field level detail.

sambhrant-metron replied:
hi @devikamehra,
We have changed as per requested.

devikamehra commented on Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml:
```diff
@@ -0,0 +1,28 @@
+triggerThreshold: 0
+query: let Votiro_view  = view () {
+  VotiroEvents
+   | where SanitizationResult == "Blocked" and PasswordProtected == "false" and From == "null"
```
Is "From" empty if no value is present or it will contain the value null always? 
Also, use =~ instead of ==, if applicable.

sambhrant-metron replied:
hi @devikamehra,
"From" will contain null value always, also we used "=~" instead of "==" where ever it was applicable.

devikamehra commented on Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml:
```diff
@@ -0,0 +1,28 @@
+query: let Votiro_view  = view () {
+  VotiroEvents
+   | where SanitizationResult == "Blocked" and PasswordProtected == "false" and From == "null"
+   | extend FileWithConnectorDetails = strcat_delim(' ', SrcFileName, 'with the hash', SrcFileSHA256, 'that was sent from connector', ConnectorName)};Votiro_view
```
File name and Hash can be entities that can be mapped.
Since we are creating Alert per result, can we summarize the data here to avoid any duplicates in a single execution?

sambhrant-metron replied:
hi @devikamehra,
We have changed as per requested.

devikamehra commented on Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml:
```diff
@@ -0,0 +1,28 @@
+name: Votiro - File Blocked from Connector
+description: |
+ 'If the syslog message includes-Blocked under  -Sanitization result-field  and false under -password protected- field and null under -from- field  create an alert with the following message:[file name] with hash [file hash] that was sent from connector [connector name] was blocked by Votiro due to Policy [policy name], see more detail in the following link [incident url]'
+severity: Low
```
The analytical rule is missing tactics and techniques.

sambhrant-metron replied:
hi @devikamehra,
We have changed as per requested.

devikamehra commented on Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml:
```diff
@@ -0,0 +1,28 @@
+query: let Votiro_view  = view () {
+  VotiroEvents
+   | where SanitizationResult == "Blocked" and PasswordProtected == "false" and From == "null"
+   | extend FileWithConnectorDetails = strcat_delim(' ', SrcFileName, 'with the hash', SrcFileSHA256, 'that was sent from connector', ConnectorName)};Votiro_view
```
Use parse-where function to ensure only required records are captured here.

sambhrant-metron replied:
hi @devikamehra,
This is a business rule, we need to get all the coloumns we used in "where" condtion along with "parse-where". We can't use more than one coloumn in parse-where operator. So we have made changes accordingly.
If you know anything else please suggest.
Thanks,
Sambhrant.

devikamehra reviewed (changes requested):
Ensure all feedbacks in first analytical rule are also considered for second one.

v-mchatla commented:
Hi @sambhrant-metron -
Can you please address the feedback given by Devika.
Thanks

sambhrant-metron commented:
Hi, @v-mchatla,
I am currently looking into the changes.
Please request you to give me sometime, as I'm verifying some.
Thanks

sambhrant-metron commented:
Hi @v-mchatla,
We have done all the changes as per request.
Please review and let me know incase of any query.
Thanks,
Sambhrant.

sambhrant-metron commented on Solutions/Votiro/Data Connectors/VotiroEvents.json:
```diff
@@ -0,0 +1,132 @@
+{
+    "id": "Votiro",
```
hi @devikamehra,
We have changed as per requested.

sambhrant-metron commented on Solutions/Votiro/Data Connectors/VotiroEvents.json:
```diff
@@ -0,0 +1,132 @@
+{
+    "id": "Votiro",
+    "title": "Votiro",
```
hi @devikamehra,
We have changed as per requested.

sambhrant-metron commented on Solutions/Votiro/Data Connectors/VotiroEvents.json:
```diff
@@ -0,0 +1,132 @@
+        {
+            "metricName": "Total data received",
+            "legend": "Votiro",
+            "baseQuery": "\nCommonSecurityLog\n| where DeviceVendor == \"Votiro\"\n"
```
hi @devikamehra,
We have changed as per requested.

v-mchatla commented:
Hi @devikamehra,
Author has made the requested changes. Can you please review the changes and share your thoughts.
Thanks

devikamehra commented on Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml:
```diff
@@ -0,0 +1,42 @@
+   | parse-where SanitizationResult with "Blocked" blocked
+   | where PasswordProtected == "false" and From =~ "null"
+   | extend FileWithConnectorDetails = strcat_delim(' ', SrcFileName, 'with the hash', SrcFileSHA256, 'that was sent from connector', ConnectorName)
+   | distinct FileWithConnectorDetails};Votiro_view
```
This will fail because you have distinct FileWithConnectorDetails at the end.

Use the function "summarize" to resolve previous comment, not "duplicate".

sambhrant-metron replied:
Hi @devikamehra,
I have used 'summarize' function instead of 'distinct'.
Please review and let me know incase of any query.
Thanks.

v-mchatla commented:
Hi @sambhrant-metron,
Please address the suggested changes.
Thanks

v-mchatla commented:
Hi @sambhrant-metron,
Sorry for the multiple follow-ups. It would be great if you could address Devika's comments.
Thanks

sambhrant-metron commented:
Hi @v-mchatla,
We are working on the request and sorry for the delay.
Thanks

sambhrant-metron commented:
Hi @v-mchatla,
Sorry for the delay.
We have made changes as per the request.
Thanks.

v-mchatla commented:
Hi @sambhrant-metron,
Thanks for making requested changes.
@devikamehra - can you please review the latest changes and share your feedback.
Thanks

saurabh-metron commented:
Hi @devikamehra, @v-mchatla,
Could you please help us push this PR forward as this was raised in December 2023 and we would like this to be available for the customers as soon as possible, thanks in advance!

devikamehra commented on Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml:
```diff
@@ -0,0 +1,42 @@
+query: let Votiro_view  = view () {
+  VotiroEvents
+   | parse-where SanitizationResult with "Blocked" blocked
+   | where PasswordProtected == "false" and From =~ "null"
```
Use =~ instead of ==

sambhrant-metron replied:
Hi @devikamehra,
We tried =~ as you suggested but while testing we faced an error showing:
![image (1)](https://user-images.githubusercontent.com/111038486/227932383-550e1e38-8b33-4278-b209-6e85955ba634.png)
So we are not making any changes to this request.
Thanks,
Sambhrant.

devikamehra commented on Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml:
```diff
@@ -0,0 +1,42 @@
+   | parse-where SanitizationResult with "Blocked" blocked
+   | where PasswordProtected == "false" and From =~ "null"
+   | extend FileWithConnectorDetails = strcat_delim(' ', SrcFileName, 'with the hash', SrcFileSHA256, 'that was sent from connector', ConnectorName)
+   | summarize count() by FileWithConnectorDetails};Votiro_view
```
We have to summarize by all relevant fields. We require SrcFileName, SrcFileSHA256, FileWithConnectorDetails, PolicyName, IncidentURL, SanitizationResult and LogSeverity for mapping.
If we summarize with only one value, we might not get the desired results and the field mapping would essentially fail.

sambhrant-metron replied:
Hi @devikamehra,
We have added all the required fields.
Please let us know incase of any query.
Thanks,
Sambhrant.

v-mchatla commented:
Hi @sambhrant-metron,
Apologies for the delay. Please request for review once you are done with the requested changes, so that reviewer can start reviewing the changes.
@devikamehra - Can you please review and provide your approval.
Thanks

sambhrant-metron commented:
Hi @v-mchatla, @devikamehra,
Please review the changes.
Let me know incase of any query.
Thanks,
Sambhrant.

v-atulyadav commented:
Hi @sambhrant-metron,
Please address the suggested changes.
Thanks

sambhrant-metron commented on Solutions/Votiro/Analytic Rules/VotiroFileBlockedInEmail.yaml:
```diff
@@ -0,0 +1,41 @@
+  - T1566
+query: let Votiro_view  = view () {
+  VotiroEvents
+   | parse-where SanitizationResult with "Blocked" blocked
```
hi @devikamehra,
We cannot use the "has" function with the "parse-where" operator.
The "parse-where" operator is used to extract fields from the raw data using regular expressions and apply filtering conditions to those fields. The output of the "parse-where" operator is a set of fields that are derived from the raw data.
On the other hand, the "has" function is used to check if a field contains a particular value. It operates on a field that already exists in the data and cannot be used to filter fields that are derived from the raw data using the "parse-where" operator.
If we want to use 'has' function we have to use 'where' instead of 'parse-where'.
Thanks,
Sambhrant.

sambhrant-metron commented:
Hi @v-atulyadav,
I have commented the query asked by @devikamehra .
Please let me know incase of any issue.
Thanks,
Sambhrant.

v-mchatla commented:
Hi @devikamehra
Can you please review the changes and provide your approval.
Thanks

sambhrant-metron commented:
Hi @v-mchatla,
We are looking into these requests. 
Thanks,
Sambhrant.

v-atulyadav commented:
Hi @sambhrant-metron, please check comments from @devikamehra and act accordingly. Thanks

sambhrant-metron commented:
Hi @v-atulyadav,
Thanks for the reminder.
We are looking into these requests.
Thanks,
Sambhrant.

v-atulyadav commented:
Hi @sambhrant-metron, 
Thank you for acknowledging. We look forward to hearing back from you.

v-atulyadav commented:
Hi @sambhrant-metron,
We look forward to hearing back from you.
Thanks

v-atulyadav commented:
Hi @sambhrant-metron,
any update?

sambhrant-metron commented:
ji @v-atulyadav,
Apologies for the delay in response.
We are still working on it and we shall update them by mid next week.
Thanks,
Sambhrant

v-atulyadav commented:
Hi @sambhrant-metron,
Thank you for acknowledging. We look forward to hearing back from you.

v-atulyadav commented:
Hi @sambhrant-metron, thanks for your acknowledgement.

suraj-metron commented on Solutions/Votiro/Workbooks/Votiro Monitoring Dashboard.json:
```diff
@@ -0,0 +1,213 @@
+      "type": 3,
+      "content": {
+        "version": "KqlItem/1.0",
+        "query": "VotiroEvents\n| where TimeGenerated {TimeRange}\n| summarize Count=count() by SanitizationResult\n| where SanitizationResult != \"\"\n| where SanitizationResult in (\"Blocked\", \"Sanitized\", \"Partially sanitized\", \"Skipped\")\n",
```
Removed the repeated condition

suraj-metron commented on Solutions/Votiro/Workbooks/Votiro Monitoring Dashboard.json:
```diff
@@ -0,0 +1,213 @@
+      "type": 3,
+      "content": {
+        "version": "KqlItem/1.0",
+        "query": "VotiroEvents\n| where TimeGenerated {TimeRange}\n| summarize Count = count() by ConnectorType\n| where ConnectorType == \"File connector\" or ConnectorType == \"Email connector\"",
```
Done

suraj-metron commented on Solutions/Votiro/Parser/VotiroEvents.txt:
```diff
@@ -0,0 +1,5 @@
+﻿CommonSecurityLog
+| parse-kv AdditionalExtensions as (CompanyName: string, CorrelationId:guid, ItemId: guid, SrcFileName: string, SrcFileSize: int, PasswordProtected: bool, AVResult: string, ThreatCount: int, BlockedCount: int, FileModification: string, SanitizationResult: string, SanitizationTime: int, ConnectorType: string, ConnectorName: string, ConnectorId: dynamic, PolicyName: string, ExceptionId: dynamic, In ...
```
Added filter for Voiro events as DeviceVendor and DeviceProduct

suraj-metron commented on Solutions/Votiro/Analytic Rules/VotiroFileBlockedFromConnector.yaml:
```diff
@@ -0,0 +1,42 @@
+  - T0837
+query: let Votiro_view  = view () {
+  VotiroEvents
+   | parse-where SanitizationResult with "Blocked" blocked
```
Hi @devikamehra ,
In this comment, we have been advised to use parse-where instead of "==": https://github.com/Azure/Azure-Sentinel/pull/6905#discussion_r1116723629

Could you please let us know which type of query condition should we use here?
Thanks

devikamehra replied:
@suraj-metron I meant to suggest changes for the last like where you were using strcat_delim.
The where clause was absolutely correct.

sambhrant-metron replied:
Hi @devikamehra,
Can you please provide any information on how can I change the strcat_delim part to parse_where.
Thanks

v-atulyadav commented:
Hi @sambhrant-metron, let us know once PR is ready to review. Thanks

suraj-metron commented:
Hi @v-atulyadav 
We have fixed the review comments.
However, we did had a doubt for one of the review comments and we have reached out to @devikamehra for the same. Waiting for her response.
The comment can be found here: https://github.com/Azure/Azure-Sentinel/pull/6905#discussion_r1158332663

Apart from the above, all the other comments are fixed and the PR is updated.

Thanks

sambhrant-metron commented:
Hi @v-atulyadav,
We have added the a new "threats" in Votiro parser.
Also it would be helpfull if you can please provide any information on how can I change the strcat_delim part to parse_where
as pointed by @devikamehra here https://github.com/Azure/Azure-Sentinel/pull/6905/#discussion_r1202774445,
we would be greatfull to you.
Thanks and Regards,
Sambhrant.

devikamehra commented:
> Hi @v-atulyadav We have fixed the review comments. However, we did had a doubt for one of the review comments and we have reached out to @devikamehra for the same. Waiting for her response. The comment can be found here: [#6905 (comment)](https://github.com/Azure/Azure-Sentinel/pull/6905#discussion_r1158332663)
> 
> Apart from the above, all the other comments are fixed and the PR is updated.
> 
> Thanks

I looked through the query again and I believe parse-where approach will not work. Please the below query.

VotiroEvents
   | where sanitizationResult has "Blocked" and passwordProtected == "false" and from =~ "null"
   | extend FileWithConnectorDetails = strcat_delim(' ', fileName, 'with the hash', SrcFileSHA256, 'that was sent from connector', connectorName)
   | summarize count() by fileName, SrcFileSHA256, FileWithConnectorDetails, policyName, tostring(incidentURL), sanitizationResult, LogSeverity};Votiro_view

v-atulyadav commented:
Hi @sambhrant-metron, @devikamehra shared a query with you please act accordingly. Thanks

sambhrant-metron commented:
Hi @v-atulyadav @devikamehra, 
Thank you so much for providing us the relevant query.
We will add it and update the PR as soon as possible.
Thanks and regards,
Sambhrant.

devikamehra commented on Solutions/Votiro/Analytic Rules/VotiroFileBlockedInEmail.yaml:
```diff
@@ -0,0 +1,40 @@
+  - T1486
+  - T1566
+query: let Votiro_view  = view () {
+  VotiroEvents
```
Same logic is used in both the queries. What is the difference here.
Also, there is no entity mapping present here which is ideal to be used in Analytical Rule.

sambhrant-metron replied:
Hi @devikamehra ,
The functionality of both is similar, except for the fields after "extend" as the data retrieval looks different.

As for the "entity mapping" part we are looking into it.
Thanks,
Sambhrant

v-mchatla commented:
Hi @sambhrant-metron 
Thanks for making requested changes, I will get it reviewed and share the feedback.
Thanks

sambhrant-metron commented:
Hi @v-mchatla ,
Please expect a delay as all the required review checks have not passed.
I will ping you as soon as all the checks have been passed.
Thanks

sambhrant-metron commented:
Hi @devikamehra ,
Please review the latest changes after adding the entity mapping
Please revert incase of any issues.
Thanks and Regards,
Sambhrant.

rahul0216 reviewed (commented):
Thanks @sambhrant-metron for the updates. Rules and Parsers look good. I have one query though,
Both the rules will trigger for same condition with almost same results. How having 2 rules for same condition is beneficial?
Also, aggregationKind is AlertPerResult, so if there are 10 results, 10 Incidents will be created. Hope you understand this.

rahul0216 commented:
@v-mchatla I have reviewed Parser and Analytic Rule. They look good. Please review/get reviewed other contents.

v-atulyadav commented:
Hi @sambhrant-metron, please check above comments from @v-mchatla and act accordingly. Thanks

sambhrant-metron commented:
hi @v-atulyadav,
I am working on the changes and commit the changes as soon as possible.
Thanks.

v-atulyadav commented:
Noted @sambhrant-metron, thanks.

v-mchatla commented:
Hi @sambhrant-metron 
I hope you are doing good. May I know the expected date for publishing the requested changes.
Thanks

sambhrant-metron commented:
Hi @v-mchatla , 
We have completed all the requested changes and updates. Following are the changes we have done:

- Changed Connectivity criteria query in data connector checking for last 30days data to 3 days.
- Workbook query for Incoming Traffic(Data Source), Added where condition before the summarize operation
- Workbook query for Scanned Files(Threats), Added where condition before the summarize operation
- Repackaged the solution with Version 3.0.0

Please let us know incase of any issues.
Thanks and Regards,
Sambhrant.

sambhrant-metron commented:
Hi @v-mchatla ,
I have setup a call today at 2:30 pm IST, if you have issues with the time schedule please suggest a new time for the meeting.
Thanks
Sambhrant.

v-mchatla commented:
Hi @sambhrant-metron 
Thanks for your time over call.
As discussed, We are going to merge this PR and applying below changes on top of your changes.
1.	Workbook metadata changes
2.	Parser file extension change from .txt to yaml
3.	Release Note
4.	Repackaging the solution with 3.0.0
5.	Removing the existing 2.0.0 version from the master and from the 3.0.0 zip
**Note:** Please don’t publish the package until we completed the remaining changes and confirmed, as we can’t publish the same version again.
Thanks
